src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControl.java

```diff
@@ -11,75 +10,102 @@
         this.roleAssignmentsFilteringService = roleAssignmentsFilteringService;
-        this.fakeRoleAssignmentsGenerator = fakeRoleAssignmentsGenerator;
+        this.pseudoRoleAssignmentsGenerator = pseudoRoleAssignmentsGenerator;
         this.applicationParams = applicationParams;
         this.accessProfileService = accessProfileService;
+        this.pseudoRoleToAccessProfileGenerator = pseudoRoleToAccessProfileGenerator;
     }
 
     // Returns Optional<CaseDetails>. If this is not enough think of wrapping it in a AccessControlResponse
     // that contains the additional information about the operation result
     @Override
     public Optional<CaseDetails> applyAccessControl(CaseDetails caseDetails) {
         RoleAssignments roleAssignments = roleAssignmentService.getRoleAssignments(securityUtils.getUserId());
-        CaseDetails cloned = caseService.clone(caseDetails);
 
         RoleAssignmentFilteringResult filteringResults = roleAssignmentsFilteringService
             .filter(roleAssignments, caseDetails);
 
-        if (filteringResults.hasGrantTypeExcludedRole()) {
-            filteringResults = filteringResults.retainBasicAndStandardGrantTypeRolesOnly();
-        }
-
         if (applicationParams.getEnablePseudoRoleAssignmentsGeneration()) {
-            List<RoleAssignment> augmentedRoleAssignments = fakeRoleAssignmentsGenerator
-                .addFakeRoleAssignments(filteringResults);
+            List<RoleAssignment> pseudoRoleAssignments = pseudoRoleAssignmentsGenerator
+                .createPseudoRoleAssignments(filteringResults);
+            filteringResults = augment(filteringResults, pseudoRoleAssignments);
         }
 
         CaseTypeDefinition caseTypeDefinition = caseTypeService.getCaseType(caseDetails.getCaseTypeId());
 
+        if (applicationParams.getEnablePseudoAccessProfilesGeneration()) {
+            List<RoleToAccessProfileDefinition> generatedAccessProfiles =
+                pseudoRoleToAccessProfileGenerator.generate(caseTypeDefinition);
+
+            List<RoleToAccessProfileDefinition> roleToAccessProfiles = caseTypeDefinition.getRoleToAccessProfiles();
+            roleToAccessProfiles.addAll(generatedAccessProfiles);
+            caseTypeDefinition.setRoleToAccessProfiles(roleToAccessProfiles);
+        }
+
+        if (filteringResults.hasGrantTypeExcludedRole()) {
+            filteringResults = filteringResults.retainBasicAndSpecificGrantTypeRolesOnly();
+        }
+
         List<AccessProfile> accessProfiles = accessProfileService
             .generateAccessProfiles(filteringResults, caseTypeDefinition);
 
-        // 4.) determine AccessProfiles from the new RoleToAccessProfiles Tab
-        // https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1460559903#AccessControlScopeofDelivery-NewRoleToAccessProfilesTab
-        // as a result we identify the AccessProfiles that the user has on the case
-
+        CaseDetails cloned = caseService.clone(caseDetails);
```

Comment:
how come this cloning?

Reply:
The flow is not closed in this class, hence this cloned instance is hanging unchanged. Moved it close to the usage, because the checkstyle was failing the build. We have initially said we would like to clone the caseDetails and apply new Access Control. Maybe we could remove it totally?

Reply:
I see. You can leave it for now, I have the suspicion that as part of the ticket Kiran is working on, he'll change the signature of this method. He will remove this if needed